yesod-core/test/Hierarchy.hs

```diff
@@ -9,6 +9,8 @@
 {-# LANGUAGE TypeSynonymInstances #-}
 {-# LANGUAGE ViewPatterns #-}
 
+{-# OPTIONS_GHC -Wno-orphans #-}
```

Comment:
testing note for a future PR: I'm wondering if it might be helpful to have snapshot/golden tests for the TH output of running the generator; not to catch bugs, but for documentation of how it works.

One thing I had trouble with reading this PR is understanding what the generated code looks like, and I think this could help with that.

Reply:
It doesn't help that the generator is highly indirected and the generated code is really difficult to read! Part of my work here is separating the parser and dispatch code so the parse code at least can be straightforward. The dispatch code may need to remain complex, but I think I can make it nicer - the `MkDispatchSettings` record is exported fully open, so it's gonna be a breaking change major bump, *but* I think this is going to be very low impact, and anyone truly impacted can help fix it up.